Detectors/TOF/compression/src/Compressor.cxx

```diff
@@ -0,0 +1,1496 @@
+{
+  if (decoderOpen(inFileName))
+    return true;
+  if (encoderOpen(outFileName))
+    return true;
+  return false;
+}
+
+bool Compressor::close()
+{
+  if (decoderClose())
+    return true;
+  if (encoderClose())
+    return true;
+  return false;
+}
+
+bool Compressor::decoderInit()
+{
+#ifdef DECODER_VERBOSE
+  if (mDecoderVerbose) {
+    std::cout << colorBlue
+              << "--- INITIALISE DECODER BUFFER: " << mDecoderBufferSize << " bytes"
+              << colorReset
+              << std::endl;
+  }
+#endif
+  if (mDecoderBuffer && mOwnDecoderBuffer) {
+    std::cout << colorYellow
+              << "--- a buffer was already allocated, cleaning"
+              << colorReset
+              << std::endl;
+    delete[] mDecoderBuffer;
+  }
+  mDecoderBuffer = new char[mDecoderBufferSize];
+  mOwnDecoderBuffer = true;
+  return false;
+}
+
+bool Compressor::encoderInit()
+{
+#ifdef ENCODER_VERBOSE
+  if (mEncoderVerbose) {
+    std::cout << colorBlue
+              << "--- INITIALISE ENCODER BUFFER: " << mEncoderBufferSize << " bytes"
+              << colorReset
+              << std::endl;
+  }
+#endif
+  if (mEncoderBuffer && mOwnEncoderBuffer) {
+    std::cout << colorYellow
+              << "-W- a buffer was already allocated, cleaning"
+              << colorReset
+              << std::endl;
+    delete[] mEncoderBuffer;
+  }
+  mEncoderBuffer = new char[mEncoderBufferSize];
+  mOwnEncoderBuffer = true;
+  encoderRewind();
+  return false;
+}
+
+bool Compressor::decoderOpen(const std::string name)
+{
+  if (mDecoderFile.is_open()) {
+    std::cout << colorYellow
+              << "-W- a file was already open, closing"
+              << colorReset
+              << std::endl;
+    mDecoderFile.close();
+  }
+  mDecoderFile.open(name.c_str(), std::fstream::in | std::fstream::binary);
+  if (!mDecoderFile.is_open()) {
+    std::cerr << colorRed
+              << "-E- Cannot open input file: " << name
+              << colorReset
+              << std::endl;
+    return true;
+  }
+  return false;
+}
+
+bool Compressor::encoderOpen(const std::string name)
+{
+  if (mEncoderFile.is_open()) {
+    std::cout << colorYellow
+              << "-W- a file was already open, closing"
+              << colorReset
+              << std::endl;
+    mEncoderFile.close();
+  }
+  mEncoderFile.open(name.c_str(), std::fstream::out | std::fstream::binary);
+  if (!mEncoderFile.is_open()) {
+    std::cerr << colorRed << "-E- Cannot open output file: " << name
+              << colorReset
+              << std::endl;
+    return true;
+  }
+  return false;
+}
+
+bool Compressor::decoderClose()
+{
+  if (mDecoderFile.is_open()) {
+    mDecoderFile.close();
+    return false;
+  }
+  return true;
+}
+
+bool Compressor::encoderClose()
+{
+  if (mEncoderFile.is_open())
+    mEncoderFile.close();
+  return false;
+}
+
+bool Compressor::decoderRead()
+{
+  if (!mDecoderFile.is_open()) {
```

Comment:
Hi Ruben,

thanks, likely we can replace the input file handling with the `RawFileReader` you mentioned.
It is not 100% clear to me if we will need `Compressor.cxx` and `Compressor.h` for standalone applications in future (not depending from O2 / DLP). But I believe I can replace it safely.

Re. `RawFileWriter` it would be very nice to have something similar. The current implementation was just a way to pass data from one DLP layer to another, for testing purposes. We are currently using for QC tests, but we are aware it was just a way to hack it.
Please, let us know when `RawFileWriter` is available and if you need more feedback from us.

Cheers,
Roberto

Reply:
OK, hope to have it this week.

Reply:
hi @shahor02 , to clarify on the use of this method.
This is not being used in the actual data-flow, but was only used for dedicated applications where we wanted to read from a file for test purposes.
Cheers,
Roberto